backend/internal/application/store.go

```diff
@@ -109,7 +109,7 @@ func (st *applicationStore) CreateApplication(app model.ApplicationProcessedDTO)
 
 // GetTotalApplicationCount retrieves the total count of applications from the database.
 func (st *applicationStore) GetTotalApplicationCount() (int, error) {
-	dbClient, err := provider.GetDBProvider().GetDBClient("identity")
```

Comment:
Shall we move this change to separate PR, while keeping the this PR only for the introduction of the user datasource?